tests/PHPStan/Analyser/data/bug-7698.php

```diff
@@ -25,13 +25,13 @@ function foo()
 {
 	$t = new Test(new A());
 	$class = $t->value::class;
-	assertType('class-string<Bug7698\A>|class-string<Bug7698\B>', $class);
+	assertType("'Bug7698\\\\A'|'Bug7698\\\\B'", $class);
```

Comment:
Don't we loose knowledge here, that the type involved is not just a regular string but a class-string?